sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ContainerThroughputControlGroupProperties.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.throughputControl;
+
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.throughputControl.config.ThroughputControlGroupInternal;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public class ContainerThroughputControlGroupProperties {
+    private static Logger logger  = LoggerFactory.getLogger(ContainerThroughputControlGroupProperties.class);
+
+    private final AtomicReference<ThroughputControlGroupInternal> defaultGroup;
+    private final Set<ThroughputControlGroupInternal> throughputControlGroupSet;
+    private final Set<String> supressInitErrorGroupSet;
+
+
+    public ContainerThroughputControlGroupProperties() {
+        this.defaultGroup = new AtomicReference<>();
+        this.throughputControlGroupSet = ConcurrentHashMap.newKeySet();
+        this.supressInitErrorGroupSet = ConcurrentHashMap.newKeySet();
+    }
+
+    public int addThroughputControlGroup(ThroughputControlGroupInternal group) {
```

Comment:
Can we please document the return behavior of this API ?
That after adding, it returns the new size of the group.

Reply:
adding javadoc comments for all public APIs here would be useful anyway

Reply:
nvm - it is in the implementation package